snap/quota/resources.go

```diff
@@ -53,12 +64,35 @@ func (qr *Resources) validateMemoryQuota() error {
 	return nil
 }
 
+func (qr *Resources) validateCpuQuota() error {
+	// if cpu count is non-zero, then percentage should be set
+	if qr.CPU.Count != 0 && qr.CPU.Percentage == 0 {
+		return fmt.Errorf("cannot validate quota limits with count of >0 and percentage of 0")
+	}
+
+	// atleast one cpu limit value must be set
+	if qr.CPU.Count == 0 && qr.CPU.Percentage == 0 && len(qr.CPU.AllowedCPUs) == 0 {
+		return fmt.Errorf("cannot validate quota limits with a cpu quota of 0 and no allowed cpus")
+	}
+	return nil
+}
+
+func (qr *Resources) validateThreadQuota() error {
+	// make sure the thread count is not zero
```

Comment:
it should also be greater than zero right?

Reply:
yes, definately!